File: 1512-number-of-good-pairs/1512-number-of-good-pairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function(nums) {
    const object = {};
    let ans = 0;
    
    for(const num of nums){
        if(object[num]){
            ans += object[num];
            object[num] += 1
        }
        else{
            object[num] = 1;
        }
    }
    
    return ans;
};

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        int ans = 0;
        unordered_map<int,int> mp;
        
        for(auto num : nums){
            ans += mp[num];
            mp[num]++;
        }
        
        return ans;
    }
};